src/libcore/num/mod.rs

```diff
@@ -464,6 +452,30 @@ macro_rules! uint_impl {
                     v => Some(self / v),
                 }
             }
+
+            #[inline]
+            fn pow(self, mut exp: u32) -> Self {
+                let mut base = self;
+                let mut acc: Self = Int::one();
+                let mut did_base_overflow = false;
+
+                while exp > 0 {
+                    if (exp & 1) == 1 {
+                        acc = acc * base;
+                        if did_base_overflow {
+                            panic!("overflow detected");
+                        }
```

Comment:
@GuillaumeGomez I fear I do not understand your question.

It seems like you are asking "why do we care about the returned result in the cases where we panic?"  When we panic, of course there is no returned result.  I care about enforcing the following rule: panic if *and only if* the accumulated `acc` (the "to-be-returned result") has been corrupted by an overflow.  If we panic, then I want to to be because the accumulated value has become useless.  I do not want to panic merely because the computation `base * base` overflows when its resulting value is never actually used.

From @alexcrichton 's comment, it sounds like he is assuming that your code was written in the context of a version of Rust that panics on overflow from operations like `base * base`.  (While on the other hand, you have written your code to still panic on overflow even in a context where `base * base` itself does not.)  If you look at my gist, it deliberately uses a version of multiply for the `base * base` squaring computation at the end of each iteration that does *not* cause a panic when that squaring overflows.  You would need to change your code to also use that variant if you want to ensure that the `base * base` computation does not itself immediately panic.